packages/core/core/src/Validation.js

```diff
@@ -0,0 +1,99 @@
+// @flow strict-local
+import nullthrows from 'nullthrows';
+import type {AssetRequest, ParcelOptions} from '@parcel/types';
+
+import path from 'path';
+import Cache from '@parcel/cache';
+import {resolveConfig} from '@parcel/utils';
+
+import type Config from './public/Config';
+import {report} from './ReporterRunner';
+import InternalAsset from './Asset';
+import type {NodeId, ConfigRequest} from './types';
+import {MutableAsset} from './public/Asset';
+import summarizeRequest from './summarizeRequest';
+
+export type ValidationOpts = {|
+  request: AssetRequest,
+  loadConfig: (ConfigRequest, NodeId) => Promise<Config>,
+  parentNodeId: NodeId,
+  options: ParcelOptions
+|};
+
+export default class Validation {
+  request: AssetRequest;
+  configRequests: Array<ConfigRequest>;
+  loadConfig: ConfigRequest => Promise<Config>;
+  options: ParcelOptions;
+  cache: Cache;
+  impactfulOptions: $Shape<ParcelOptions>;
+
+  constructor({request, loadConfig, parentNodeId, options}: ValidationOpts) {
+    this.request = request;
+    this.configRequests = [];
+    this.loadConfig = configRequest => {
+      this.configRequests.push(configRequest);
+      return loadConfig(configRequest, parentNodeId);
+    };
+    this.options = options;
+  }
+
+  async run(): Promise<void> {
+    if (this.request.filePath.includes('node_modules')) return;
```

Comment:
Should we just not queue files in node_modules at all? Would avoid crossing the IPC boundary just to do nothing...

Reply:
This was how it was originally but you commented on it to move it to the validatorrunner.

Will change it back. Well it wasn't a queue back then...

Reply:
Oh.. haha sorry!

Reply:
feel free to update in a followup